src/components/user-view/document-timeline/timeline-footer.tsx

```diff
@@ -81,6 +205,7 @@ export default function TimelineFooter(props: {
  exists, it calculates whether the content inside the element overflows horizontally. */
   useEffect(() => {
     if (footerTimelineRef.current === null) return;
+    dispatch(loadTimelinePoints({ timelinePoints }));
```

Comment:
2 things:

- we shouldn't ever call dispatch from within a functional component. Anything like this should be moved into a hook function (e.g. use-with-state.tsx) 
- it looks like you're adding timelinePoints into the redux state just to be able to check its length in the QontoStepIcon component, which is an over-implementation. You should be able to just pass down this info from one of its parent components that already has the timelinePoints data (looks like the TimelineFooter component has this data)

Reply:
It looks like it does not accept props from parent component. I opted for another solution much more easies only using icons.